### ISSUE TITLE
Operation returned an invalid status code 'Forbidden

### ISSUE THREAD

**Issue by sidsajjan1** [Azure/ipam#251]
Hi There
I'm seeking some guidance to resolve the issue.

I am a owner and global administrator of the azure tenant. I am receiving a error during the execution as noted below, however, IPAM deployment succeeds and I'm unable to view the subscription under IPAM Application

----------------
Deployment logs:

New-AzRoleAssignment: /Users/xxxxxxx/.local/share/powershell/Modules/Az.Resources/6.16.0/MSGraph.Autorest/custom/New-AzADServicePrincipal.ps1:752:13
Line |
 752 |        $ra = New-AzRoleAssignment @param
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Operation returned an invalid status code 'Forbidden'

-----------
Log streams under web apps:

024-03-21T06:14:43.402704667Z Exception Details:	(AccessDenied) Access is denied to the requested resource. The user might not have enough permission.
2024-03-21T06:14:43.402731267Z 	Code: AccessDenied
2024-03-21T06:14:43.402737167Z 	Message: Access is denied to the requested resource. The user might not have enough permission.
2024-03-21T06:14:43.402742467Z 2024-03-21 06:14:43.401 | ERROR    | app.main:find_reservations:480 - Error running network check loop!

---------------
Detailed deployment logs
--------------
NOTE: IPAM Deployment Type: AppContainer
INFO: Fetching Tenant ID from Azure PowerShell SDK
INFO: Fetching Azure Cloud type from Azure PowerShell SDK
INFO: Validating Azure Region selected for deployment
INFO: Azure Region validated successfully
INFO: Creating Azure IPAM UI Application
INFO: Creating Azure IPAM Engine Application
INFO: Updating Azure IPAM Engine API Endpoint
INFO: Updating Azure IPAM UI Application Resource Access
INFO: Creating Azure IPAM UI Service Principal
INFO: Creating Azure IPAM Engine Service Principal
New-AzRoleAssignment: /Users/xxxxxxx/.local/share/powershell/Modules/Az.Resources/6.16.0/MSGraph.Autorest/custom/New-AzADServicePrincipal.ps1:752:13
Line |
 752 |        $ra = New-AzRoleAssignment @param
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Operation returned an invalid status code 'Forbidden'
INFO: Creating Azure IPAM Engine Secret
INFO: Azure IPAM Engine & UI Applications/Service Principals created successfully
INFO: Logging in to Microsoft Graph
INFO: Granting admin consent for Microsoft Graph API permissions assigned to IPAM UI application
INFO: Admin consent for Microsoft Graph API permissions granted successfully
INFO: Granting admin consent to the IPAM UI application for exposed API from the IPAM Engine application
INFO: Admin consent for IPAM Engine exposed API granted successfully
INFO: Granting admin consent for Azure Service Management API permissions assigned to IPAM Engine application
INFO: Admin consent for Azure Service Management API permissions granted successfully
INFO: Deploying IPAM bicep templates
INFO: IPAM bicep templates deployed successfully
INFO: Updating UI Application with SPA configuration
INFO: UI Application SPA configuration update complete
INFO: Azure IPAM Solution deployed successfully

Thanks in advance

**Comment by DCMattyG** [comment]
Hi @sidsajjan1, this is already documented in our Guide here:

https://azure.github.io/ipam/#/troubleshooting/README?id=new-azroleassignment-failed

Essentially you are missing the required permissions under the Root Tenant Management Group as described in the Prerequisites which can be found here:

https://azure.github.io/ipam/#/deployment/README?id=prerequisites

Hope that helps!

**Comment by DCMattyG** [comment]
Hi there @sidsajjan1, just wanted to circle back and make sure that the details I shared above were helpful. If not please let me know and we'll do what it takes to make sure you're up and running.

Thanks for reaching out!

**Comment by sidsajjan1** [comment]
Hi @DCMattyG Thank you for your quick responses and sorry for the delay replying to the thread.

I have reviewed  tenant level permissions. 
Since this was my dev/test tenant, I missed configuring ENTRAID properties to 
Access management for Azure resources that my account can manage access to all Azure subscriptions and management groups in this tenant. Once I set this to YES. 
IPAM is working as expected :)

ref: https://www.tunecom.be/quick-tip-reset-access-to-your-root-management-group/

Kind regards
